src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -223,6 +229,145 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
+                new Object[]{new HaplotypeProbabilitiesFromContaminatorSequence(hb, .1, 5, 0, 1), new HaplotypeProbabilitiesFromContaminatorSequence(hb, .1, 5, 1, 0)},
+        };
+    }
+
+    @Test(dataProvider = "mergeIsDafeProvider")
+    public void testMergeHaplotypeProbabilitiesIsSafe(final HaplotypeProbabilities hp1, final HaplotypeProbabilities hp2){
+
+        final HaplotypeProbabilities merged1 = hp1.deepCopy().merge(hp2);
+        final HaplotypeProbabilities merged2 = hp1.deepCopy().merge(hp2);
+
+        Assert.assertEquals(merged1.getLikelihoods(),merged2.getLikelihoods());
+    }
+
+
+    @Test(dataProvider = "mergeIsDafeProvider")
+    public void testMergeFingerprintIsSafe(final HaplotypeProbabilities hp1, final HaplotypeProbabilities hp2){
+
+        final Fingerprint fpA = new Fingerprint("test2",null,"none");
+        final Fingerprint fpB = new Fingerprint("test2",null,"none");
+
+        final Fingerprint fp1 = new Fingerprint("test1",null,"none");
+        fp1.add(hp1);
+
+        final Fingerprint fp2 = new Fingerprint("test1",null,"none");
+        fp2.add(hp2);
+
+        fpA.merge(fp1);
+        fpB.merge(fp1);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertEquals(fpA.get(hb).getLikelihoods(),fpB.get(hb).getLikelihoods());
+        }
+
+        fpA.merge(fp2);
+        fpB.merge(fp2);
+        fpB.merge(fp2);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertNotEquals(fpA.get(hb), fpB.get(hb));
+        }
+
+        fpA.merge(fp2);
+
+        for(HaplotypeBlock hb:fpA.keySet()){
+            Assert.assertEquals(fpA.get(hb).getLikelihoods(),fpB.get(hb).getLikelihoods());
+        }
+    }
+
+
+    @Test
+    public void testMergeIsSafeFromSequence() {
+        final Path na12891_r1 = new File(TEST_DATA_DIR, "NA12891.over.fingerprints.r1.sam").toPath();
+        final Path na12891_r2 = new File(TEST_DATA_DIR, "NA12891.over.fingerprints.r2.sam").toPath();
+        final Path na12892_r1 = new File(TEST_DATA_DIR, "NA12892.over.fingerprints.r1.sam").toPath();
+        final Path na12892_r2 = new File(TEST_DATA_DIR, "NA12892.over.fingerprints.r2.sam").toPath();
+
+        final List<Path> listOfFiles = Arrays.asList(na12891_r1, na12891_r2, na12892_r1, na12892_r2);
+
+        final FingerprintChecker checker = new FingerprintChecker(SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING);
+
+        final Map<FingerprintIdDetails, Fingerprint> fingerprintIdDetailsFingerprintMap = checker.fingerprintFiles(listOfFiles, 1, 0, TimeUnit.DAYS);
+
+        final Fingerprint combinedFp = new Fingerprint("test", null, null);
+        fingerprintIdDetailsFingerprintMap.values().forEach(combinedFp::merge);
+
+        final Fingerprint combinedFp2 = new Fingerprint("test2", null, null);
+        fingerprintIdDetailsFingerprintMap.values().forEach(combinedFp2::merge);
+
+        for (final HaplotypeBlock block : combinedFp.keySet()) {
```

Comment:
In general, if you have an assert in a `for`, you need to assert that the thing you're looping over has at least one item, otherwise the test will pass when the `for` never executes.